### PR TITLE
TRT-10 GA Support for main branch

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

This PR runs CI to test TRT-10 GA support for main branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ These are the following dependencies used to verify the testcases. Torch-TensorR
 - Libtorch 2.4.0.dev (latest nightly) (built with CUDA 12.1)
 - CUDA 12.1
 - cuDNN 8.9.5
```
Comment (peri044) [inline README.md:121]: Can you remove this line ?

**Review by peri044 — CHANGES_REQUESTED** [review]
1) Can you add tensorrt==10.0.1.6 in our py/requirements.txt ? (https://github.com/pytorch/TensorRT/blob/trt_10_ga_cloned_from_main/py/requirements.txt)
2) Also update trt version in dev_dep_versions.yml (https://github.com/pytorch/TensorRT/blob/trt_10_ga_cloned_from_main/dev_dep_versions.yml)

**Review by peri044 — APPROVED** [review]
Changes LGTM. pending rebase

File: dev_dep_versions.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 __version__: "2.3.0.dev0"
```
Comment (narendasan) [inline dev_dep_versions.yml:1]: 2.4.0.dev0

File: py/requirements.txt [hunk 3]
```diff
@@ -1,9 +1,9 @@
 numpy
 packaging
 pybind11==2.6.2
---extra-index-url https://download.pytorch.org/whl/nightly/cu121
-torch>=2.4.0.dev,<2.5.0
-torchvision>=0.19.0.dev,<0.20.0
+--extra-index-url https://download.pytorch.org/whl/cu121
+torch==2.3.0
```
Comment (narendasan) [inline py/requirements.txt:5]: This should not be changed for main

File: pyproject.toml [hunk 4]
```diff
@@ -8,7 +8,7 @@ requires = [
     "cffi>=1.15.1",
     "typing-extensions>=4.7.0",
     "future>=0.18.3",
-    "tensorrt==10.0.0b6",
+    "tensorrt==10.0.1.6",
```
Comment (HolyWu) [inline pyproject.toml:11]: The version of TRT python package is `10.0.1`, no trailing `.6`.

**Comment by HolyWu** [comment]
https://github.com/pytorch/TensorRT/blob/20b34e24d7ab285c069c6aac37bca382dcbd18ec/pyproject.toml#L45 also needs to specify version.

**Comment by zewenli98** [comment]
@HolyWu Made the changes. Thanks for the review!

File: dev_dep_versions.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
 __version__: "2.4.0.dev0"
 __cuda_version__: "12.1"
 __cudnn_version__: "8.9"
-__tensorrt_version__: "10.0.0.6"
+__tensorrt_version__: "10.0.1.6"
```
Comment (HolyWu) [inline dev_dep_versions.yml:4]: Oops. I overlooked this.

File: tests/py/dynamo/models/test_dyn_models.py [hunk 6]
```diff
@@ -66,6 +67,7 @@ def forward(self, x):
         torch.cuda.empty_cache()
 
 
+@unittest.skip("Skipping this test for now due to a bug")
```
Comment (narendasan) [inline tests/py/dynamo/models/test_dyn_models.py:70]: Can you add the reason for the failure to this message?

Comment (zewenli98) [reply tests/py/dynamo/models/test_dyn_models.py:70]: I opened an issue for the bug: https://github.com/pytorch/TensorRT/issues/2794

**Pull request merged**